### PR TITLE
Appropriate suggestions

### LINKED ISSUE #165: Interaction suggestions depending on own status
## User Story

**As a** user -  
**I want to** only receive suggestions for users I might call on the dashboard tab if I have set the status "Available" or "Nice2MeetU" -  
**In order to** avoid distractions when I'm busy. 

## Conditions of satisfaction

- [x] if I have not set an appropriate status, I don't get any suggestions (see wireframe)
- [ ] if I have set an appropriate status, I receive up to 4 suggestions **of users which have set a status of available or Nice2MeetU** with one of them standing out (see wireframe)

## Implementation Note

Don't care too much about card design here, this is subject of another issue.

## Wireframe

image

### PR DESCRIPTION
**Pull request by frcroth**
Closes #165

### CODE REVIEW

**Comment by frcroth** [comment]
Implemented no suggestions on working:
image

File: app/models/user.rb [hunk 1]
```diff
@@ -85,6 +85,10 @@ def name
     "#{firstname} #{lastname}"
   end
 
+  def unavailable
```
Comment (felixauringer) [inline app/models/user.rb]:
```suggestion
  def unavailable?
```
Methods returning a boolean normally end with a question mark :)

File: spec/features/dashboard/user_list_spec.rb [hunk 2]
```diff
@@ -26,6 +26,23 @@
     end
   end
 
+  context 'when in working status' do
+    let!(:available_user) { FactoryBot.create :user, current_status: User.filter_status }
+
+    before do
+      user.current_status = 'working'
```
Comment (felixauringer) [inline spec/features/dashboard/user_list_spec.rb]:
```suggestion
      user.current_status = User.status_working
```

File: app/views/home/_users.html.erb [hunk 3]
```diff
@@ -5,22 +5,28 @@
...
-        <span><i class="fa fa-user"></i></span>
-      </div>
-      <div class="col user-name">
-        <%= user.display_name %>
-      </div>
-      <div class="user-meeting d-flex">
-        <%= form_with model: JitsiCall.new, id: "init-call-#{user.id}" do |f| %>
-          <%= f.hidden_field :guest_id, value: user.id %>
-          <%= button_tag(type: 'submit', class: "fa fa-phone fa-flip-horizontal phone-button") do %>
+  <% if current_user.unavailable %>
```
Comment (felixauringer) [inline app/views/home/_users.html.erb]:
```suggestion
  <% if current_user.unavailable? %>
```

**Review by felixauringer — APPROVED** [review]

**Comment by felixauringer** [comment]
For this to close #165 the second point (`if I have set an appropriate status, I receive up to 4 suggestions of my contacts which have set a status of available or Nice2MeetU with one of them standing out (see wireframe)`) is still missing, right?

**Comment by frcroth** [comment]
> 
> 
> For this to close #165 the second point (`if I have set an appropriate status, I receive up to 4 suggestions of my contacts which have set a status of available or Nice2MeetU with one of them standing out (see wireframe)`) is still missing, right?

Yes, that's why it is a draft

**Comment by felixauringer** [comment]
> Yes, that's why it is a draft

Ups, didn't see that, sorry :sweat_smile:

**Comment by frcroth** [comment]
image

File: app/views/home/_users.html.erb [hunk 4]
```diff
@@ -5,22 +5,33 @@
...
+      <p><%= I18n.t('user.status.busy_text') %></p>
+    </div>
+  <% else %>
+    <% @users.each do |user| %>
+      <% if user == @users[0] %>
+        <h4><%= I18n.t('user.status.start_talk') %></h4>
+      <% end %>
+      <% if user == @users[1] %>
+        <h4><%= I18n.t('user.status.or_maybe') %></h4>
+      <% end %>
```
Comment (felixauringer) [inline app/views/home/_users.html.erb]:
```suggestion
    <% @users.each_with_index do |user, index| %>
      <% if index == 0 %>
        <h4><%= I18n.t('user.status.start_talk') %></h4>
      <% end %>
      <% if index == 1 %>
        <h4><%= I18n.t('user.status.or_maybe') %></h4>
      <% end %>
```

After writing this I'm not so sure anymore that this is an improvement :sweat_smile:  What do you think @frcroth ?

**Review by felixauringer — APPROVED** [review]
One could add test for the two headings you added, but it's fine without them for me.

**Review by felixauringer — APPROVED** [review]

**Comment by frcroth** [comment]
There are still some things to do:

- [x] Limit to 4
- [x] Make available a possible status
Optional

- [x] Use user cards for the suggestions

**Comment by frcroth** [comment]
Current progress: 
image

**Comment by frcroth** [comment]
image
I updated the styling a little. It doesn't have to be perfect, because #189 will probably change all that.

File: app/controllers/home_controller.rb [hunk 5]
```diff
@@ -26,6 +26,6 @@ def chat
   private
 
   def maximum_length_user_list
-    30
+    4
   end
```
Comment (sirkrypt0) [inline app/controllers/home_controller.rb]: Could this be a variable like we have done with the `VALID_STATUS_LIST`?

File: app/models/user.rb [hunk 6]
```diff
@@ -42,7 +42,7 @@ def self.select_status_list
   end
 
   def self.filter_status
-    status_nice_to_meet_you
+    [status_nice_to_meet_you, status_available]
```
Comment (sirkrypt0) [inline app/models/user.rb:45]: Could this be a variable like we have done with the `VALID_STATUS_LIST`?

File: app/views/users/_card.html.erb [hunk 7]
```diff
@@ -1,11 +1,11 @@
-<div class="card shadow rounded-lg">
-  <div class="card-body">
-    <div class="container">
+<div class="user-card card shadow rounded-lg">
+  <div class="card-body user-card-body">
+    <div class="container" style="padding-top: 5px;padding-bottom: 5px;">
```
Comment (sirkrypt0) [inline app/views/users/_card.html.erb:3]: Uh, inline style, this should be in a CSS file imo.

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 6 above)
Comment (frcroth) [reply app/models/user.rb:45]: I think this is already enough abstraction

**Review by sirkrypt0 — APPROVED** [review]

**Pull request merged**